dbms/src/Storages/KVStore/Decode/PartitionStreams.cpp

```diff
@@ -69,30 +69,12 @@ struct AtomicReadWriteCtx
         , table_id(table_id_)
     {}
 
-    std::optional<Block> tryUseDecodeCache(const RegionPtrWithBlock & region, ManageableStoragePtr & storage)
+    std::optional<Block> tryUseDecodeCache(const RegionPtrWithBlock &, ManageableStoragePtr &)
     {
-        if unlikely (region.pre_decode_cache)
-        {
-            auto schema_version = storage->getTableInfo().schema_version;
-            std::stringstream ss;
-            region.pre_decode_cache->toString(ss);
-            LOG_DEBUG(
-                log,
-                "{} got pre-decode cache {}, storage schema version: {}",
-                region->toString(),
-                ss.str(),
-                schema_version);
-
-            if (region.pre_decode_cache->schema_version == schema_version)
-            {
-                return std::move(region.pre_decode_cache->block);
-            }
-            else
-            {
-                LOG_DEBUG(log, "schema version not equal, try to re-decode region cache into block");
-                region.pre_decode_cache->block.clear();
-            }
-        }
+        // Currently, RegionPtrWithBlock with a not-null CachePtr is only used in debug functions
+        // to apply a pre-decoded snapshot. So it will not take place here.
+        // we keep this function here as a tombstone, so we don't have to bisect some codes to
+        // find out why some codes are missing.
```

Comment:
IIMO we don't need to keep legacy code/function to increase the code complexity. Or the code will be rot as time go by.

Reply:
OK, I will remove the comments.